### PR TITLE
Add HTML repr to `LabelledDict`

### PR DESCRIPTION
**Pull request by h-mayorquin**
## Motivation

Right now it looks like this:
image

After this PR:
image

## Checklist

- [x] Did you update `CHANGELOG.md` with your changes?
- [x] Does the PR clearly describe the problem and the solution?
- [x] Have you reviewed our [Contributing Guide](https://github.com/hdmf-dev/hdmf/blob/dev/docs/CONTRIBUTING.rst)?
- [x] Does the PR use "Fix #XXX" notation to tell GitHub to close the relevant issue numbered XXX when the PR is merged?

### CODE REVIEW

**Comment by rly** [comment]
Can you add tests for the branches (empty case and nested case)? Otherwise looks good to me

**Comment by rly** [comment]
I can add this into the 4.3.1 release today if it's ready today.

**Comment by h-mayorquin** [comment]
> I can add this into the 4.3.1 release today if it's ready today.

Let me add the tests

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # HDMF Changelog
 
+## HDMF 4.3.1 (Upcoming)
```
Comment (h-mayorquin) [inline CHANGELOG.md:3]: So we probably want to move this if this is being released today, right?

**Review by rly — APPROVED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (rly) [reply CHANGELOG.md:3]: This is fine. I'll update the changelog to the appropriate date in a separate PR.
